version: '3.7'

services:
  nginx:
    build: ./nginx
    ports:
      - 443:443
    volumes:
      - static_volume:/var/www/python_django__togobi/static
    depends_on:
      - web
    restart: "on-failure"

  web:
    build: .
    command: sh -c "python manage.py runserver 0.0.0.0:${APP_PORT}"
    volumes:
      - .:/var/www/python_django__togobi/:rw
      - static_volume:/var/www/python_django__togobi/static
    env_file:
      - .env
    image: python_django__togobi
    expose:
      - ${APP_PORT}
    depends_on:
      - db_postgres
    restart: "on-failure"
    tty: true

  db_postgres:
    image: postgres:11-alpine
    volumes:
      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
      - postgres_data:/var/lib/postgresql/data/
    environment:
      - POSTGRES_PASSWORD=${DATABASE_PASSWORD}
      - POSTGRES_DB=${DATABASE_NAME}
      - PGPORT=${DATABASE_PORT}
      - POSTGRES_USER=${DATABASE_USER}
    ports:
      - "6543:5432"
    restart: "on-failure"

  pgadmin:
      image: dpage/pgadmin4
      restart: always
      environment:
        PGADMIN_DEFAULT_EMAIL: admin@admin.com
        PGADMIN_DEFAULT_PASSWORD: root
      ports:
        - "5050:80"

  db_redis:
    image: "redis:6-alpine"


volumes:
  postgres_data:
  static_volume: